src/NuGet.Clients/PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -271,7 +267,10 @@ private void HandleItemLoaderStateChange(IItemLoader<PackageItemListViewModel> l
 
                     if (!Items.Contains(_loadingStatusIndicator))
                     {
-                        Items.Add(_loadingStatusIndicator);
+                        lock (_itemsLock)
```

Comment:
yes, but right when your if condition evaluates to true in this case, another thread could go ahead and add the loadingstatus indicator right before the lock is acquired .

---
In reply to: [75760740](https://github.com/NuGet/NuGet.Client/pull/838#discussion_r75760740) [](ancestors = 75760740)